cuda_core/tests/test_context.py

```diff
@@ -3,8 +3,88 @@
 
 import cuda.core.experimental
 import pytest
+from cuda.core.experimental import Device
 
 
 def test_context_init_disabled():
     with pytest.raises(RuntimeError, match=r"^Context objects cannot be instantiated directly\."):
         cuda.core.experimental._context.Context()  # Ensure back door is locked.
+
+
+# ============================================================================
+# Context Equality Tests
+# ============================================================================
```

Comment:
It would probably be difficult, but is there any way we can add a test for when we have different contexts and/or an invalid context in some way?